crypto/evp/signature.c

```diff
@@ -25,15 +25,21 @@ static EVP_SIGNATURE *evp_signature_new(OSSL_PROVIDER *prov)
     if (signature == NULL)
         return NULL;
 
+    if (!CRYPTO_NEW_REF(&signature->refcnt, 1)) {
+        OPENSSL_free(signature);
+        return NULL;
+    }
+
     signature->lock = CRYPTO_THREAD_lock_new();
```

Comment:
I don't see anywhere that we still need this lock? Am I missing it?

Reply:
You're right.  I must have missed this one when I went through each removing the lock and checking for failures.